Completion/Linux/Command/_nmcli

```diff
@@ -1,244 +1,369 @@
-#compdef nmcli
+#compdef _nmcli nmcli
 
-local context state line expl
-typeset -A opt_args
+# compatible with nmcli 1.8.4
 
-if [[ -z $_nmcli_version ]]; then
-    _nmcli_version="${"$(_call_program nmcli $words[1] --version)"##*version }"
-fi
+_nmcli() {
+    local line    _arguments -C \
+        {'(-terse)-t','(-t)-terse'}"[terse output]" \
+        {'(-pretty)-p','(-p)-pretty'}"[pretty output]" \
+        {'(-mode)-m','(-m)-mode'}"[output mode]: :(tabular multiline)" \
+        {'(-colors)-c','(-c)-colors'}"[whether to use colors in output]: :(auto yes no)" \
+        {'(-fields)-f','(-f)-fields'}"[specify fields to output]:fields:_nmcli_fields" \
+        {'(-get-values)-g','(-g)-get-values'}"[shortcut for -m tabular -t -f]:fields:_nmcli_fields" \
+        {'(-escape)-e','(-e)-escape'}"[escape columns separators in values]: :(yes no)" \
+        {'(-ask)-a','(-a)-ask'}"[ask for missing parameters]" \
+        {'(-show-secrets)-s','(-s)-show-secrets'}"[allow displaying passwords]" \
+        {'(-wait)-w','(-w)-wait'}'[set timeout waiting for finishing operations]:timeout (s)' \
+        {'(-version)-v','(-v)-version'}"[show program version]" \
+        {'(-help)-h','(-h)-help'}"[print help]" \
+        "1: :(general networking radio connection device agent monitor help)" \
+        "*::arg:->args"
 
-(( $+functions[_nmcli_is_running] )) || _nmcli_is_running() {
-    [[ $(_call_program nmcli $words[1] -f running nm) != *'not running'* ]]
+    case $line[1] in
+        g*) _nmcli_general ;;
+        n*) _nmcli_networking ;;
+        r*) _nmcli_radio ;;
+        c*) _nmcli_connection ;;
+        d*) _nmcli_device ;;
+        a*) _nmcli_agent ;;
+    esac
 }
 
-(( $+functions[_nmcli_con_ids] )) || _nmcli_con_ids() {
-    _nmcli_is_running || return
+_nmcli_fields() {
+    _values -s , 'fields to be printed' \
```

Comment:
By convention, description should be singular, i.e. "field". Only one is completed at a time even if a list is allowed.